app/dashboard/models.py

```diff
@@ -2028,6 +2033,23 @@ def point_value(self):
     def i18n_name(self):
         return _(next((x[1] for x in self.ACTIVITY_TYPES if x[0] == self.activity_type), 'Unknown type'))
 
+    @property
+    def text(self):
+        from django.template.loader import render_to_string
+        from bs4 import BeautifulSoup
+        params = {
+            'row': self,
+            'hide_date': True,
+            'hide_likes': True,
+        }
+        html_str = render_to_string('shared/activity.html', params)
+        soup = BeautifulSoup(html_str)
```

Comment:
Whats the advantage of using BS here? I'm thinking we can do `html_str = html_str.strip("\n").replace("  ", " ")`but maybe im missing the context behind the for loop here

Reply:
it removes all HTML, not just newlines